Don't fail on values that contain variables

bastelfreak opened the pull request:
Due to the way puppet-lint parses values that contain strings, we need to implement an exception. See
https://puppet-lint.com/developer/tokens/#strings

bastelfreak committed:
Enhance tests from slack usecases

kenyon commented on spec/puppet-lint/plugins/check_params_empty_string_assignment/check_params_empty_string_assignment_spec.rb:
```diff
@@ -108,6 +108,74 @@ class foo ( String $var = 'public' ) { }
+        <<-EOS
+        class foo (
+          String $install_path = 'bla',
+          String $bin_path = ${install_path}/public",
```
Missing opening quotation mark?

bastelfreak replied:
upsi, fixed

bastelfreak committed:
Don't fail on values that contain variables

Due to the way puppet-lint parses values that contain strings, we need
to implement an exception. See
https://puppet-lint.com/developer/tokens/#strings
